tests/integration_tests_rust.rs

```diff
@@ -28,42 +28,57 @@ use std::sync::Arc;
 #[test]
 fn channel_full_cycle() {
 	let (bitcoind, electrsd) = setup_bitcoind_and_electrsd();
-	let (node_a, node_b) = setup_two_nodes(&electrsd, false, true, false);
+	let chain_source = TestChainSource::Esplora(&electrsd);
+	let (node_a, node_b) = setup_two_nodes(&chain_source, false, true, false);
+	do_channel_full_cycle(node_a, node_b, &bitcoind.client, &electrsd.client, false, true, false);
+}
+
+#[test]
+fn channel_full_cycle_bitcoind() {
+	let (bitcoind, electrsd) = setup_bitcoind_and_electrsd();
+	premine_dummy_blocks(&bitcoind.client, &electrsd.client);
+	let chain_source = TestChainSource::BitcoinRpc(&bitcoind);
+	let (node_a, node_b) = setup_two_nodes(&chain_source, false, true, false);
 	do_channel_full_cycle(node_a, node_b, &bitcoind.client, &electrsd.client, false, true, false);
 }
```

Comment:
Why not expand the other tests for `BitcoinRpc`?

Reply:
Hmm... probably only tests where non-setup code depends on block (dis)connection notification (e.g., `onchain_spend_receive`). Though, I also wonder if we should use `BitcoinRpc` normally just to remove one moving part from the tests assuming that would lead to less flakiness.

Reply:
Right, if the extra part is actually removing the underlying flakiness, maybe we should keep it. 😛 Though perhaps we need to make sure the (non-test) code is robust enough to retry appropriately.